src/core/infer_request.cc

```diff
@@ -199,7 +307,7 @@ InferenceRequest::Release(
   // and the callback may interact with upper level trace.
   if (request->trace_ != nullptr) {
     request->trace_->ReportNow(TRITONSERVER_TRACE_REQUEST_END);
-    InferenceTrace::Release(std::move(request->trace_));
+    request->trace_ = nullptr;
   }
```

Comment:
Also need to call
response_factory_.SetTrace(nullptr) or the equivalent to drop the trace reference that is held by the response factory.

Reply:
change to call request->ReleaseTrace